internal/controllers/clusterextension_controller.go

```diff
@@ -330,16 +372,28 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	return ctrl.Result{}, nil
 }
 
-func (r *ClusterExtensionReconciler) resolve(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, error) {
+type ResolveErrorType int
+
+const (
+	PackageNotFoundError ResolveErrorType = iota
+	VersionNotFoundError
+	OtherResolveError
+)
+
+func (r ResolveErrorType) String() string {
+	return [...]string{"PackageNotFound", "VersionNotFound", "Other"}[r]
+}
+
+func (r *ClusterExtensionReconciler) resolve(ctx context.Context, clusterExtension ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, ResolveErrorType, error) {
```

Comment:
To minimize changes, can you rename `clusterExtension` to `ext`?

Reply:
And I still don't understand the need for `ResolveErrorType`, lint is complaining about it.

Reply:
Refactor done.

lint is complaining b/c nothing uses the package not found err type. I'd like to work it out so this is used and we can report properly that package is non existent after resolve. And then I think the extra errorType will make sense.

Reply:
Perhaps use aggregate errors instead?, so no new return argument is needed?
e.g. 
```
var PackageNotFoundError = fmt.Errorf("package not found")
var VersionNotFoundError = fmt.Errorf("version not found")
var OtherResolveError = fmt.Errorf("other resolve error")

...

    // in resolve()
    return nil, utilerrors.NewAggregate([]error{OtherResolveError, err})

...
    // in reconcile():
    switch {
        case agg.Is(OtherResolveError): 
            ....
        case agg.Is(VersionNotFoundError):
            ...
        case agg.Is(OtherResolveError):
            ...
        default:
            ...
    }

Reply:
think latest commit refactors out the added type and moves it all to use the aggregate error.